crates/solver/src/liquidity/order_converter.rs

```diff
@@ -85,23 +87,18 @@ struct OrderSettlementHandler {
 
 /// Returns (`sell_amount`, `fee_amount`) for the given order and adjusts the values accordingly
 /// for limit orders.
-fn compute_synthetic_order_amounts_for_limit_order(order: &Order) -> Result<(U256, U256)> {
-    anyhow::ensure!(
-        order.metadata.class == OrderClass::Limit,
-        "this function should only be called for limit orders"
-    );
-    let surplus_fee = order
-        .metadata
-        .surplus_fee
-        .context("limit order does not have surplus fee set")?;
+fn compute_synthetic_order_amounts_for_limit_order(
+    order: &Order,
+    limit: &LimitOrderClass,
+) -> Result<(U256, U256)> {
```

Comment:
Not sure about this one.
On the one hand it's nice that you have the `LimitOrderClass` which requires the existence of a limit order to call this function. On the other hand you could still pass in a market order here.
If `compute_synthetic_order_amounts_for_limit_order()` were a function on the `Order` itself we wouldn't have this problem but this functionality is pretty specific to the order encoding.
All the options don't make me very happy but I also don't have a better idea. 🤷‍♂️ 

Same issue with `custom_price_for_limit_order()`.